src/DependencyManagement/DependencyManager.cs

```diff
@@ -1,393 +1,245 @@
+            IDependencySnapshotInstaller installer = null)
         {
-            Dependencies = new List<DependencyInfo>();
+            _moduleProvider = moduleProvider ?? new PowerShellGalleryModuleProvider();
+            _storage = storage ?? new DependencyManagerStorage(GetFunctionAppRootPath(requestMetadataDirectory));
+            _installedDependenciesLocator = installedDependenciesLocator ?? new InstalledDependenciesLocator(_storage);
+            _installer = installer ?? new DependencySnapshotInstaller(_moduleProvider, _storage, new DependencySnapshotPurger(_storage));
         }
 
         /// <summary>
-        /// Processes the dependency download request
+        /// Initializes the dependency manager:
+        /// - Parses functionAppRoot\requirements.psd1 file and creates a list of dependencies to install.
+        /// - Determines the snapshot directory path to use.
         /// </summary>
-        /// <param name="msgStream">The protobuf messaging stream</param>
-        /// <param name="request">The StreamingMessage request for function load</param>
-        /// <param name="pwsh">The PowerShell instance used to download modules</param>
-        internal void ProcessDependencyDownload(MessagingStream msgStream, StreamingMessage request, PowerShell pwsh)
+        /// <returns>
+        /// The dependency snapshot path where all the required dependencies are installed
+        /// or will be installed. This path can be added to PSModulePath.
+        /// Returns null if managed dependencies are disabled or the 
+        /// </returns>
+        public string Initialize(StreamingMessage request, ILogger logger)
         {
-            if (request.FunctionLoadRequest.ManagedDependencyEnabled)
+            if (!request.FunctionLoadRequest.ManagedDependencyEnabled)
             {
-                var rpcLogger = new RpcLogger(msgStream);
-                rpcLogger.SetContext(request.RequestId, null);
+                return null;
+            }
 
-                if (!_shouldUpdateFunctionAppDependencies)
-                {
-                    if (!string.IsNullOrEmpty(_dependenciesNotUpdatedMessage))
-                    {
-                        // We were not able to update the function app dependencies.
-                        // However, there is a previous installation, so continue with the function app execution.
-                        rpcLogger.Log(LogLevel.Warning, _dependenciesNotUpdatedMessage, isUserLog: true);
-                    }
-                    else
-                    {
-                        // The function app already has the latest dependencies installed.
-                        rpcLogger.Log(LogLevel.Trace, PowerShellWorkerStrings.LatestFunctionAppDependenciesAlreadyInstalled, isUserLog: true);
-                    }
+            return Initialize(logger);
+        }
 
-                    return;
-                }
+        internal string Initialize(ILogger logger)
```

Comment:
I would really love to avoid documenting all methods, especially private and internal. If you feel otherwise, let's discuss this offline.

For this specific method, there is a public entry point with different parameters (`Initialize(StreamingMessage request, ILogger logger)`) and this one is documented.